app/src/main/java/ch/hikemate/app/utils/MapUtils.kt

```diff
@@ -25,18 +25,19 @@ object MapUtils {
    * Shows a hike on the map.
    *
    * @param mapView The map view where the hike will be shown.
-   * @param hike The hike to be shown.
+   * @param waypoints The points that compose the line to show on the map.
    * @param color The color of the hike.
+   * @param onLineClick To be called when the line on the map is clicked.
    */
   fun showHikeOnMap(
       mapView: MapView,
-      hike: HikeRoute,
+      waypoints: List<LatLong>,
       color: Int,
       onLineClick: () -> Unit,
   ) {
     val line = Polyline()
 
-    line.setPoints(hike.ways.map { GeoPoint(it.lat, it.lon) })
+    line.setPoints(waypoints.map { GeoPoint(it.lat, it.lon) })
     line.outlinePaint.color = color
     line.outlinePaint.strokeWidth = MapScreen.STROKE_WIDTH
```

Comment:
There were two reasons behind these changes:

- `showHikeOnMap` does not need anything else than the waypoints, so the idea was to only request what is required
- We could pass the hike, but now the waypoints of the hike are of type `DeferredData`, so it would be a pain to handle this in `showHikeOnMap`. It's easier to handle it before and once we have them call the function.

Reply:
Good second point. Then we can keep it like this.